tests/js/client/shell/shell-hotbackup-cluster-nonwindows.js

```diff
@@ -0,0 +1,118 @@
+/*jshint globalstrict:false, strict:false */
+/*global arango, assertTrue, assertEqual, assertNotEqual, db*/
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test hotbackup in single server
+///
+/// Copyright 2024 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// @author Michael Hackstein
+////////////////////////////////////////////////////////////////////////////////
+
+const jsunity = require('jsunity');
+const serverHelper = require('@arangodb/test-helper');
+const {readAgencyValueAt} = require("@arangodb/testutils/replicated-logs-helper");
+const {debugSetFailAt, debugCanUseFailAt, debugClearFailAt, errors} = require('internal');
+
+'use strict';
+
+function HotBackupSuite () {
+
+  return {
+    testRegressionCanRestoreBackupWithMissingAnalyzerEntry: function() {
+      const dbName = "otherDB";
+      const colName = "otherCollection";
+      const colNameTwo = "additionalCollection";
+      const dropExtraCollections = () => {
+        try {
+          db._dropDatabase(dbName);
+        } catch (e) {}
+        db._drop(colName);
+      };
+
+      const assertAdditionalDataDoesNotExist = () => {
+        assertEqual(db._databases().indexOf(dbName), -1, "Setup issue: Database was not dropped before restore");
+        assertEqual(db._collections().map(c => c.name()).indexOf(colName), -1, "Setup issue: Collection was not dropped before restore");
+        assertNotEqual(db._collections().map(c => c.name()).indexOf(colNameTwo), -1, "Collection was not created between backup and restore");
+      };
+      const assertAdditionalDataExists = () => {
+        assertNotEqual(db._databases().indexOf(dbName), -1, "Database was not restored");
+        assertNotEqual(db._collections().map(c => c.name()).indexOf(colName), -1, "Collection was not restored");
+        assertEqual(db._collections().map(c => c.name()).indexOf(colNameTwo), -1, "After Backup Collection was not erased");
+      };
+      try {
+        db._createDatabase(dbName);
+        db._create(colName);
+
+        const backupGood = arango.POST("/_admin/backup/create", {}).result;
+
+        // Now create the regression. In version 3.6 the entry /arango/Plan/Analyzers did not exist.
+
+
+        assertTrue(readAgencyValueAt("Plan/Analyzers") !== undefined, "We should have analyzers in plan by default.");
```

Comment:
nit: instead of `assertTrue(x !== undefined)` we could use `assertNotUndefined(x)`, which is more legible.

Reply:
Ah nice, did not know this existed!